src/python-bindings/classad_parsers.cpp

```diff
@@ -269,6 +269,9 @@ OldClassAdIterator::next()
                     m_source.attr("seek")(end_ptr);
                     PyErr_Restore(ptype, pvalue, ptraceback);
                 }
+#if PY_MAJOR_VERSION >= 3
+                PyErr_Clear();
```

Comment:
See the commit message in 80b65605a.  Also, the surrounding code appeared to go out of its way to preserve the error state, so in case that was ever relevant, I left that behavior for python 2.  But otherwise I have no problem with clearing it in both python 2 & 3.